iac-extensions/arm/src/main/java/org/sonar/iac/arm/parser/bicep/TreeFactory.java

```diff
@@ -132,6 +136,20 @@ public SeparatedList<Expression, SyntaxToken> functionCallArguments(Expression f
     return separatedList(firstArgument, additionalArguments);
   }
 
+  public ForExpression forExpression(SyntaxToken rightBracket, SyntaxToken forKeyword, ForVariableBlock forVariableBlock,
+    SyntaxToken inKeyword, Expression expression, SyntaxToken colon, Expression forBody, SyntaxToken leftBracket) {
```

Comment:
This is a code smell, cause there are too many arguments. IMO this is not a problem. WDYT?

Reply:
I agree, to avoid it we would have to split which IMO would reduce readability. Let's keep it like this.

Reply:
BTW I believe you inverted `rightBracket` and `leftBracket` variable name :D

Reply:
Yes, good catch!